tests/rptest/tests/cluster_metrics_test.py

```diff
@@ -49,7 +49,7 @@ def controller_stable():
             nonlocal prev
             curr = self.redpanda.controller()
 
-            if prev != curr:
+            if curr is None or prev != curr:
```

Comment:
It's a pretty probabilistic test: there is a chance that `redpadna.controller()` hits the same node asking for a controller and got the same answer while the other nodes think otherwise. Please use `admin.py::await_stable_leader` it asks all the nodes and get more stable result.